core/src/org/pentaho/platform/engine/core/system/objfac/spring/BarrierBeanPostProcessor.java

```diff
@@ -32,6 +34,8 @@
 public class BarrierBeanPostProcessor implements BeanPostProcessor {
   static IServiceBarrierManager serviceBarrierManager;
```

Comment:
This class needs a little work. It's a Singleton, but has a mix of static and non-static members